packages/sdk-react/src/AdheseSlot.tsx

```diff
@@ -62,21 +69,53 @@ export function AdheseSlot({
     }) satisfies AdheseSlotOptions['setup'], [setup, onChange]),
   });
 
+  if (!lazyLoading && (slotState?.status === 'loading' || slotState?.status === 'initialized' || slotState?.status === 'initializing')) {
+    return placeholder && (
+      <SlotWrapper slot={slotState} {...props}>
+        {placeholder}
+      </SlotWrapper>
+    );
+  }
+
   if (lazyLoading || (slotState?.status === 'loaded' || slotState?.status === 'rendered' || slotState?.status === 'rendering')) {
     return (
-      <div
-        ref={element}
+      <SlotWrapper
         id={componentId}
-        data-name={slotState?.name}
-        style={{
-          width: slotState?.options.width,
-          height: slotState?.options.height,
-          ...style,
-        }}
+        slot={slotState}
         {...props}
-      />
+      >
+
+        {slotState?.status === 'rendered' ? render?.(slotState) : placeholder}
```

Comment:
Could you clarify this for me, please? why we can't do 
```
if(slotState.status === 'rendered') return render

return placeholder
```

and why do we need in not lazy loading placeholder **&&** slotState? 
I'm confused

Reply:
This is indeed very confusing. I changed it to be much cleaner